src/features/messages/queries/parse.ts

```diff
@@ -37,6 +44,70 @@ export function parseMessageQueryResult(
   return queryResult(multiProvider, scrapedChains, data, parseMessage);
 }
 
+function getTokenSymbolFromWarpRouteChainAddressMap(
+  chainMetadata: ChainMetadata,
+  address: Address,
+  warpRouteChainAddressMap: WarpRouteChainAddressMap,
+) {
+  const { name } = chainMetadata;
+  if (objKeys(warpRouteChainAddressMap).includes(name)) {
+    const chain = warpRouteChainAddressMap[name];
+    if (objKeys(chain).includes(address)) {
+      return chain[address];
+    }
+  }
+
+  return undefined;
+}
+
+export function parseWarpRouteDetails(
+  message: Message,
+  warpRouteChainAddressMap: WarpRouteChainAddressMap,
+): WarpRouteDetails | undefined {
+  try {
+    const {
+      body,
+      origin: { to },
+      totalPayment,
+      originMetadata,
+      destinationMetadata,
+      recipient,
+    } = message;
+
+    if (!body || !originMetadata || !destinationMetadata) return undefined;
+
+    const originTokenSymbol = getTokenSymbolFromWarpRouteChainAddressMap(
+      originMetadata,
+      to,
+      warpRouteChainAddressMap,
+    );
+    const destinationTokenSymbol = getTokenSymbolFromWarpRouteChainAddressMap(
+      destinationMetadata,
+      recipient,
+      warpRouteChainAddressMap,
+    );
+
+    // If token symbols are not found with the addresses, it means the message is not a warp route transfer
+    if (!originTokenSymbol || !destinationTokenSymbol) return undefined;
+
+    const parsedMessage = parseWarpRouteMessage(body);
+    const address = postgresByteaToAddress(parsedMessage.recipient, destinationMetadata);
+
+    return {
+      amount: fromWei(parsedMessage.amount.toString(), originMetadata.nativeToken?.decimals || 18),
```

Comment:
I think this is incorrect. The decimals for the warp token may not match the decimals for the native token. E.g. native ETH uses 18 but USDC uses 6.
You need the warp token's decimals here, which should be in the warp config